src/server/protocol.ts

```diff
@@ -1951,6 +1958,12 @@ namespace ts.server.protocol {
      */
     export interface QuickInfoRequest extends FileLocationRequest {
         command: CommandTypes.Quickinfo;
+        arguments: QuickInfoRequestArgs;
+    }
+
+    export interface QuickInfoRequestArgs extends FileLocationRequestArgs {
+        /** if true - return response with documentation as display parts instead of string */
+        richResponse?: boolean;

```

Comment:
Unless this is expected to vary from request to request, it seems like it should be a user preference.

Reply:
Probably, all three occurrences should be covered by a single user preference.